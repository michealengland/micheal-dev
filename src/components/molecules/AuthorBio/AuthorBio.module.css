.authorBio {
  @apply flex items-center max-w-xs mr-auto bg-purple-100 text-primary-purple1 rounded-full my-8;

  & .avatar {
    @apply rounded-full mb-0 mr-4;

    max-width: 50px;
    max-height: 50px;
  }

  & .content {
    @apply mb-0 mr-4;
  }
}
