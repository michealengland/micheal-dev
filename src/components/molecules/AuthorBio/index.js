import React from 'react'
import {StaticQuery, graphql} from 'gatsby'
import Avatar from '@images/mle-12-05-21-512x512.jpeg'
import styles from './AuthorBio.module.css'

export default function AuthorBio() {
  return (
    <StaticQuery
      query={authorQuery}
      render={(data) => {
        const {author} = data.site.siteMetadata
        return (
          <div className={styles.authorBio}>
            <img className={styles.avatar} src={Avatar} alt="Micheal England" />
            <p className={styles.content}>
              Article by <strong>{author}</strong>.
            </p>
          </div>
        )
      }}
    />
  )
}

const authorQuery = graphql`
  query AuthorQuery {
    site {
      siteMetadata {
        author
        social {
          twitter
        }
      }
    }
  }
`
