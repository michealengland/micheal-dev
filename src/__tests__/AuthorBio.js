import React from 'react'
import renderer from 'react-test-renderer'
import AuthorBio from '@components/molecules/AuthorBio'
import {StaticQuery} from 'gatsby'
// Setup StaticQuery mock.
beforeEach(() => {
  StaticQuery.mockImplementationOnce(({render}) =>
    render({
	  site: {
        siteMetadata: {
		  author: 'Mike England',
		  social: {
            twitter: 'https://example.com',
		  },
        },
	  }
    })
  )
})


describe('AuthorBio', () => {
  it('renders correctly', () => {
    const data = {
      site: {
        siteMetadata: {
          author: 'Mike England',
          social: {
            twitter: 'https://example.com',
          },
        },
      }
    }
    const tree = renderer.create(<AuthorBio data={data} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
