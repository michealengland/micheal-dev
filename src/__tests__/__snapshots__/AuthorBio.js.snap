// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthorBio renders correctly 1`] = `
<div
  className="authorBio"
>
  <img
    alt="Micheal England"
    className="avatar"
    src="test-file-stub"
  />
  <p
    className="content"
  >
    Article by 
    <strong>
      Mike England
    </strong>
    .
  </p>
</div>
`;
